builder/qemu/config.go

```diff
@@ -187,6 +187,13 @@ type QemuEFIBootConfig struct {
 	//
 	// Default: `/usr/share/OVMF/OVMF_VARS.fd`
 	OVMFVars string `mapstructure:"efi_firmware_vars" required:"false"`
+	// Keep the efivars.fd file in the exported artifacts.
+	//
+	// By default this is true, and will include the generated efivars.fd file
+	// in the artifacts built from the builder.
+	// If you don't want this file to be exported as an artifact by the builder,
+	// you can explicitely set this to `false`.
+	KeepEFIVars config.Trilean `mapstructure:"efi_keep_efivars" required:"false"`
```

Comment:
From looking at the issue. It seems like the exporting of the efi vars file was an added change to the artifact. Does it make sense not to export by default and instead of keeping maybe having this be something like 
```
source "qemu" "ubuntu" {
#[...]
  By default this is false. Setting export varfiles to true will include the generated efivars.fd along with the builder artifact. 
  efi_export_varfiles = true

```

Reply:
So regarding `efivars`, this contains a bunch of variables which are useful if you've setup secure boot for example, as some data gets written to the EFI "flash" for later use (like keys for example).
That's why I opted to include it in the artifacts at the end, and why not including this in the output artifact became an option here.

Now fundamentally, I am not necessarily opposed to not including the `efivars` in the final artifacts for the build, it's just that if you're booting an EFI machine, it's generally good to include those with the built image since it keeps state, but for most cases it's not necessary.

Bit on the fence on this one then, I would argue that we should include it by default since it has its importance, but at the same time for use cases like the one described in the related issue, it may be a good idea not to include it in the final artifact (the file will still be kept locally AFAICT, just it won't be processed by post-processors).

Thoughts?

Reply:
So this is a valid thought and I don’t have a clear answer for.  I think your understanding of the problem makes  a good point on why it should be included. 

If you think of it from the post processor and builder artifact perspective as an artifact being passed to another process does it make sense to include it by default?

having the ability to export is a good call. I just don’t know enough to say if it should be on by default.